src/GraphMsg.chpl

```diff
@@ -0,0 +1,4510 @@
+
+                           proc xlocal(x :int, low:int, high:int):bool{
+                                      if (low<=x && x<=high) {
+                                          return true;
+                                      } else {
+                                          return false;
+                                      }
+                           }
+                           var switchratio=(numCurF:real)/nf.size:real;
+                           if (switchratio<GivenRatio) {//top down
+                               topdown+=1;
+                               forall i in SetCurF with (ref SetNextF) {
+                                  if ((xlocal(i,vertexBegin,vertexEnd)) ) {// current edge has the vertex
+                                      var    numNF=nf[i];
+                                      var    edgeId=sf[i];
+                                      var nextStart=max(edgeId,edgeBegin);
+                                      var nextEnd=min(edgeEnd,edgeId+numNF-1);
+                                      ref NF=df[nextStart..nextEnd];
+                                      forall j in NF with (ref SetNextF){
+                                             if (depth[j]==-1) {
+                                                   depth[j]=cur_level+1;
+                                                   SetNextF.add(j);
+                                             }
+                                      }
+                                  } 
+                                  if ((xlocal(i,vertexBeginR,vertexEndR)) )  {
+                                      var    numNF=nfR[i];
+                                      var    edgeId=sfR[i];
+                                      var nextStart=max(edgeId,edgeBegin);
+                                      var nextEnd=min(edgeEnd,edgeId+numNF-1);
+                                      ref NF=dfR[nextStart..nextEnd];
+                                      forall j in NF with (ref SetNextF)  {
+                                             if (depth[j]==-1) {
+                                                   depth[j]=cur_level+1;
+                                                   SetNextF.add(j);
+                                             }
+                                      }
+                                  }
+                               }//end coforall
+                           }else {// bottom up
+                               bottomup+=1;
+                               //var UnVisitedSet= new set(int,parSafe = true);//use set to keep the unvisited vertices
+                               //forall i in vertexBegin..vertexEnd with (ref UnVisitedSet) {
+                               //   if depth[i]==-1 {
+                               //      UnVisitedSet.add(i);
+                               //   }
+                               //}
+                               forall i in vertexBegin..vertexEnd  with (ref SetNextF) {
```

Comment:
The logical structure of this loop is the same as the one below, the only difference is the data being operated on (i.e. range and nf/sf vs nfR/sfR.  If possible this logic should be encapsulated into a single proc where the data is passed in to be operated on.

Also the commented out code should be removed.